src/internal_rules.jl

```diff
@@ -1647,3 +1647,101 @@ function EnzymeRules.reverse(
     dst::Annotation{<:AbstractArray})
     return (nothing, nothing)
 end
+
+
+_hypotforward(x::Const) = zero(x.val)
+_hypotforward(x) = real(conj(x.val) * x.dval)
+_hypotforward(x::Const, i) = zero(x.val)
+_hypotforward(x, i) = real(conj(x.val) * x.dval[i])
+
+function EnzymeRules.forward(
+        config::EnzymeRules.FwdConfig,
+        func::Const{typeof(Base.hypot)},
+        RT,
+        x::Annotation,
```

Comment:
can you change this to x, y, z, vararg. That way the usual handling still applies for 2 args or fewer

Reply:
Oh, sure.  I just thought it'd be better to include it here.  I'll change it.

Reply:
ironically no the built in rules are extremely efficient [and can interplay with optimization passes in a way that cannot be done with custom rules]